dds/DCPS/DataReaderImpl_T.h

```diff
@@ -943,17 +943,25 @@ namespace OpenDDS {
       ser.encoding(encoding);
     }
 
+    bool ser_ret = true;
     MessageType data;
     if (sample.header_.key_fields_only_) {
-      ser >> OpenDDS::DCPS::KeyOnly<MessageType>(data);
+      ser_ret = (ser >> OpenDDS::DCPS::KeyOnly<MessageType>(data));
     } else {
-      ser >> data;
+      ser_ret = (ser >> data);
     }
-    if (!ser.good_bit()) {
-      ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) %CDataReaderImpl::lookup_instance ")
-                 ACE_TEXT("deserialization failed.\n"),
-                 TraitsType::type_name()));
-      return;
+    if (!ser_ret) {
+      if (!ser.good_bit()) {
```

Comment:
It shouldn't be using `good_bit` at all. In fact nothing outside of the Serializer.* files should. If you want to distinguish construction failure you must create a separate status for it because other things can return a serialization error that don't set `good_bit`, like a "must understand" error, which has nothing to do with construction.